client/src/leap/soledad/client/encdecpool.py

```diff
@@ -619,8 +619,10 @@ def _async_decrypt_received_docs(self):
         """
         docs = yield self._get_docs(encrypted=True)
         for doc_id, rev, content, gen, trans_id, _, idx in docs:
-            self._async_decrypt_doc(
-                doc_id, rev, content, gen, trans_id, idx)
+            if doc_id not in self._decrypting_docs:
+                self._decrypting_docs.append(doc_id)
```

Comment:
Yes. Previously to the move of the creation of the encryption pool to api this was not necessary, because we had one pool for each sync. Now that the pool is shared, this is needed yes. We also have to add a way to empty the sync db async enc table periodically, in case we have outdated document revisions there that will never be synced.